### PR TITLE
Add support for eth_getTransactionCount to prover

### PR DESCRIPTION
**Pull request by nazarhussain**
**Motivation**

Add support for web3 method `eth_getTransactionCount` 

**Description**

Add support for web3 method `eth_getTransactionCount` 

Partially: #4706 

**Steps to test or reproduce**

Run all tests.

### CODE REVIEW

File: packages/prover/src/verified_requests/eth_getTransactionCount.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import {ELVerifiedRequestHandler} from "../interfaces.js";
+import {bufferToHex} from "../utils/conversion.js";
+import {getELProof, isValidAccount, isValidStorageKeys} from "../utils/execution.js";
+import {generateRPCResponseForPayload, generateUnverifiedResponseForPayload} from "../utils/json_rpc.js";
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export const eth_getTransactionCount: ELVerifiedRequestHandler<
```
Comment (twoeths) [inline packages/prover/src/verified_requests/eth_getTransactionCount.ts:7]: @nazarhussain since it shares most of the steps from `ethGetBalance`, can we refactor to use a common method? could be in a separate PR

**Review by twoeths — APPROVED** [review]

File: packages/prover/src/verified_requests/eth_getTransactionCount.ts (hunk 1 above)
Comment (nazarhussain) [reply packages/prover/src/verified_requests/eth_getTransactionCount.ts:7]: Sure will refactor in upcoming PRs.

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.8.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.8.0) :tada:
